tests/unit/anchore_engine/services/policy_engine/engine/test_vulnerability_matches.py

```diff
@@ -63,6 +78,61 @@ def vulnerability_with_fix(empty_vulnerability):
     return fixed_vuln
 
 
+@pytest.fixture
+def vulnerability_with_nofix(empty_vulnerability):
+    fixed_vuln = copy.deepcopy(empty_vulnerability)
+    f = FixedArtifact()
+    f.vulnerability_id = fixed_vuln.id
+    f.name = 'pkg1'
+    f.namespace_name = fixed_vuln.namespace_name
+    f.version = 'None'
+    f.version_format = 'RPM'
+    f.parent = fixed_vuln
+    f.include_later_versions = True
+    f.epochless_version = f.version
+    f.fix_metadata = {}
+    f.created_at = datetime.datetime.now()
+    f.updated_at = datetime.datetime.now()
+    f.fix_observed_at = f.updated_at
+    fixed_vuln.fixed_in = [f]
+    return fixed_vuln
+
+
+@pytest.fixture
+def vulnerability_with_multifix(empty_semver_vulnerability):
+    fixed_vuln = copy.deepcopy(empty_semver_vulnerability)
+    f = FixedArtifact()
+    f.vulnerability_id = fixed_vuln.id
+    f.name = 'semverpkg1'
+    f.namespace_name = fixed_vuln.namespace_name
+    f.version = '>= 1.1.0 < 1.1.2'
+    f.version_format = 'semver'
+    f.parent = fixed_vuln
+    f.include_later_versions = False
+    f.epochless_version = f.version
+    f.fix_metadata = {'first_patched_version': '1.1.2'}
+    f.created_at = datetime.datetime.now()
+    f.updated_at = datetime.datetime.now()
+    f.fix_observed_at = f.updated_at
+    fixed_vuln.fixed_in = [f]
+
+    f = FixedArtifact()
```

Comment:
this `f` instance doesn't look like it is used in `fixed_vuln`, that is probably intentional, but I am not following how that is working to get two fixed artifacts if the `empty_semver_vulnerability` is only used with `fixed_vuln`

Reply:
The parent relationship connects it up in the fixed_in collection (`f.parent = fixed_vuln`) due to how SQLAlchemy classes work with the relationship definition. Line 117 is actually redundant and can be removed.

Reply:
Ahhh that is what I was hoping for, but wasn't sure that we were dealing with a model here.